### PR TITLE
[3.5] Added ui tests

### PR DESCRIPTION
**Pull request by Seraphym-Tuhai**
## Pull Request description

Added UI smoke tests on Rule Chains and Customers  

## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [x] PR name contains fix version. For example, "[3.3.4] Hotfix of some UI component" or "[3.4] New Super Feature".
- [x] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [ ] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [ ] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [ ] Changes are backward compatible or upgrade script is provided.
- [x] Similar PR is opened for PE version to simplify merge. Required for internal contributors only.

### CODE REVIEW

File: msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/CreateRuleChainTest.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+import org.testng.annotations.Test;
+import org.thingsboard.server.msa.ui.base.AbstractDiverBaseTest;
+import org.thingsboard.server.msa.ui.pages.LoginPageHelper;
+import org.thingsboard.server.msa.ui.pages.RuleChainsPageHelper;
+import org.thingsboard.server.msa.ui.pages.SideBarMenuViewElements;
+import org.thingsboard.server.msa.ui.utils.EntityPrototypes;
+
+import java.util.ArrayList;
+
+import static org.thingsboard.server.msa.ui.utils.Const.*;
```
Comment (YevhenBondarenko) [inline msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/CreateRuleChainTest.java:31]: We do not use wildcards.

File: msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/DeleteRuleChainTest.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+import io.qameta.allure.Description;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.thingsboard.server.msa.ui.base.AbstractDiverBaseTest;
+import org.thingsboard.server.msa.ui.pages.LoginPageHelper;
+import org.thingsboard.server.msa.ui.pages.RuleChainsPageHelper;
+import org.thingsboard.server.msa.ui.pages.SideBarMenuViewElements;
+
+import static org.thingsboard.server.msa.ui.utils.Const.*;
```
Comment (YevhenBondarenko) [inline msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/DeleteRuleChainTest.java:27]: We do not use wildcards.

File: msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/DeleteSeveralRuleChainsTest.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import io.qameta.allure.Description;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.thingsboard.server.msa.ui.base.AbstractDiverBaseTest;
+import org.thingsboard.server.msa.ui.pages.LoginPageHelper;
+import org.thingsboard.server.msa.ui.pages.RuleChainsPageHelper;
+import org.thingsboard.server.msa.ui.pages.SideBarMenuViewElements;
+
+import static org.thingsboard.server.msa.ui.utils.Const.*;
```
Comment (YevhenBondarenko) [inline msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/DeleteSeveralRuleChainsTest.java:27]: We do not use wildcards.

File: msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/MakeRuleChainRootTest.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.thingsboard.server.msa.ui.base.AbstractDiverBaseTest;
+import org.thingsboard.server.msa.ui.pages.LoginPageHelper;
+import org.thingsboard.server.msa.ui.pages.RuleChainsPageHelper;
+import org.thingsboard.server.msa.ui.pages.SideBarMenuViewElements;
+
+import static org.thingsboard.server.msa.ui.utils.Const.*;
```
Comment (YevhenBondarenko) [inline msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/MakeRuleChainRootTest.java:28]: We do not use wildcards.

File: msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/OpenRuleChainTest.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.thingsboard.server.msa.ui.base.AbstractDiverBaseTest;
+import org.thingsboard.server.msa.ui.pages.LoginPageHelper;
+import org.thingsboard.server.msa.ui.pages.OpenRuleChainPageHelper;
+import org.thingsboard.server.msa.ui.pages.RuleChainsPageHelper;
+import org.thingsboard.server.msa.ui.pages.SideBarMenuViewElements;
+
+import static org.thingsboard.server.msa.ui.utils.Const.*;
```
Comment (YevhenBondarenko) [inline msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/OpenRuleChainTest.java:28]: We do not use wildcards.

File: msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/RuleChainEditMenuTest.java [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.thingsboard.server.msa.ui.base.AbstractDiverBaseTest;
+import org.thingsboard.server.msa.ui.pages.LoginPageHelper;
+import org.thingsboard.server.msa.ui.pages.RuleChainsPageHelper;
+import org.thingsboard.server.msa.ui.pages.SideBarMenuViewElements;
+
+import static org.thingsboard.server.msa.ui.utils.Const.*;
```
Comment (YevhenBondarenko) [inline msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/RuleChainEditMenuTest.java:28]: We do not use wildcards.

File: msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/SearchRuleChainTest.java [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.thingsboard.server.msa.ui.base.AbstractDiverBaseTest;
+import org.thingsboard.server.msa.ui.pages.LoginPageHelper;
+import org.thingsboard.server.msa.ui.pages.RuleChainsPageHelper;
+import org.thingsboard.server.msa.ui.pages.SideBarMenuViewElements;
+import org.thingsboard.server.msa.ui.utils.DataProviderCredential;
+
+import static org.thingsboard.server.msa.ui.utils.Const.*;
```
Comment (YevhenBondarenko) [inline msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/SearchRuleChainTest.java:28]: We do not use wildcards.

File: msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/SortByNameTest.java [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.thingsboard.server.msa.ui.base.AbstractDiverBaseTest;
+import org.thingsboard.server.msa.ui.pages.LoginPageHelper;
+import org.thingsboard.server.msa.ui.pages.RuleChainsPageHelper;
+import org.thingsboard.server.msa.ui.pages.SideBarMenuViewElements;
+import org.thingsboard.server.msa.ui.utils.DataProviderCredential;
+
+import static org.thingsboard.server.msa.ui.utils.Const.*;
```
Comment (YevhenBondarenko) [inline msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/SortByNameTest.java:29]: We do not use wildcards.

Comment (Seraphym-Tuhai) [reply msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/SortByNameTest.java]: > We do not use wildcards.

done

File: msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/SearchRuleChainTest.java (hunk 7 above)
Comment (Seraphym-Tuhai) [reply msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/SearchRuleChainTest.java]: > We do not use wildcards.

done

File: msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/RuleChainEditMenuTest.java (hunk 6 above)
Comment (Seraphym-Tuhai) [reply msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/RuleChainEditMenuTest.java]: > We do not use wildcards.

done

File: msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/OpenRuleChainTest.java (hunk 5 above)
Comment (Seraphym-Tuhai) [reply msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/OpenRuleChainTest.java]: > We do not use wildcards.

done

File: msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/MakeRuleChainRootTest.java (hunk 4 above)
Comment (Seraphym-Tuhai) [reply msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/MakeRuleChainRootTest.java]: > We do not use wildcards.

done

File: msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/SortByNameTest.java [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+            testRestClient.deleteRuleChain(getRuleChainByName(ruleChainName).getId());
+            ruleChainName = null;
+        }
+    }
+
+    @Test(priority = 10, groups = "smoke", dataProviderClass = DataProviderCredential.class, dataProvider = "nameForSort")
+    @Description
+    public void specialCharacterUp(String ruleChainName) {
+        testRestClient.postRuleChain(defaultRuleChainPrototype(ruleChainName));
+        ;
```
[the remark this answers is not in the document]
Comment (Seraphym-Tuhai) [reply msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/SortByNameTest.java:61]: > remove

done

File: msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/RuleChainEditMenuTest.java [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+            testRestClient.deleteRuleChain(getRuleChainByName(ruleChainName).getId());
+            ruleChainName = null;
+        }
+    }
+
+    @Test(priority = 10, groups = "smoke")
+    @Description
+    public void changeName() {
+        testRestClient.postRuleChain(defaultRuleChainPrototype(ENTITY_NAME));
+        ;
```
[the remark this answers is not in the document]
Comment (Seraphym-Tuhai) [reply msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/RuleChainEditMenuTest.java]: > remove

done

File: msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/CreateRuleChainImportTest.java [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+        Assert.assertNotNull(ruleChainsPage.entity(IMPORT_RULE_CHAIN_NAME));
+        Assert.assertTrue(ruleChainsPage.entity(IMPORT_RULE_CHAIN_NAME).isDisplayed());
+    }
+
+    @Test(priority = 40, groups = "smoke")
+    @Description
+    public void importRuleChainAndSaveWithSameName() {
+        ruleChainName = IMPORT_RULE_CHAIN_NAME;
+        testRestClient.postRuleChain(defaultRuleChainPrototype(ruleChainName));
+        ;
```
[the remark this answers is not in the document]
Comment (Seraphym-Tuhai) [reply msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/ruleChainsSmoke/CreateRuleChainImportTest.java]: > remove

done

File: msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/base/AbstractDiverBaseTest.java [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+
+@Slf4j
+@Listeners(TestListener.class)
+abstract public class AbstractDiverBaseTest extends AbstractContainerTest {
+
+    protected WebDriver driver;
+    private final Dimension dimension = new Dimension(WIDTH, HEIGHT);
+    private static final int WIDTH = 1680;
+    private static final int HEIGHT = 1050;
+    private static final boolean HEADLESS = false;
```
[the remark this answers is not in the document]
Comment (Seraphym-Tuhai) [reply msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/base/AbstractDiverBaseTest.java]: > headless should be true by default.

done

File: msa/black-box-tests/src/test/resources/smokesRuleChain.xml [hunk 13]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="ISO-8859-1"?>
 <!--
 
-    Copyright © 2016-2022 The Thingsboard Authors
+    Copyright � 2016-2022 The Thingsboard Authors
```
Comment (YevhenBondarenko) [inline msa/black-box-tests/src/test/resources/smokesRuleChain.xml:4]: use mvn license:format before `push`

**Review by YevhenBondarenko — APPROVED** [review]

File: msa/black-box-tests/src/test/resources/docker-selenium.yml [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+    shm_size: 2gb
+    environment:
+      SE_NODE_MAX_SESSIONS: 8
+      SE_NODE_OVERRIDE_MAX_SESSIONS: 'true'
+      SE_NODE_SESSION_TIMEOUT: 5000
+      SE_SCREEN_WIDTH: 1920
+      SE_SCREEN_HEIGHT: 1080
+      SE_SCREEN_DEPTH: 24
+      SE_SCREEN_DPI: 74
+    extra_hosts:
```
Comment (smatvienko-tb) [inline msa/black-box-tests/src/test/resources/docker-selenium.yml:34]: An explicit IP address may cause problems in different environments. You never know what IP address/network will be assigned by the host systems.

**Review by smatvienko-tb — CHANGES_REQUESTED** [review]

**Review by YevhenBondarenko — APPROVED** [review]

**Pull request merged**